[AME-1281] Add a2_ validation to account id input and rename from advertiser ID to ad account ID

elmiranda opened the pull request:
## 💸 TL;DR
<!-- What's the three sentence summary of purpose of the PR -->

Part of the advertiser id --> ad account id migration! Intent is to remove requirements that the id conforms to _only_ `t2_` ~and make sure it looks spick and span~ and allow them to be prefixed with `a2_`.

woodchuckgym commented on template.tpl:
```diff
@@ -41,19 +41,12 @@ ___TEMPLATE_PARAMETERS___
+    "displayName": "Ad Account ID",
     "simpleValueType": true,
     "name": "id",
     "type": "TEXT",
```
sn00bie question: is the trailing comma here OK?  That is, that's not valid JSON, but it is valid JavaScript.  (Was this file autogenerated, or edited by hand?)

elmiranda replied:
Autogenerated!

elmiranda replied:
Excellent question!

elmiranda replied:
Good catch, I tried the replacement again and it should be fixed now. Bad copy pasta, sorry!

woodchuckgym reviewed (approved):
TY!  One question about formatting.

dmaii reviewed (approved):
Assuming that this was tested in GTM and it's ok with the JSON, LGTM!

elmiranda commented:
Hey folks, I'm retagging because I've updated (and rebased) this PR via https://github.com/reddit/reddit-gtm-template/pull/18/commits/37011455c86074ae4e9caa2225bd1e8d753b2542; this commit is basically the entire change set. I'll deploy once it's reapproved.

P.S. I told gym I would do this like 2 weeks ago but I got locked out of my Github account! Whoopie! We gucci now.

woodchuckgym commented on template.tpl:
```diff
@@ -44,16 +44,16 @@ ___TEMPLATE_PARAMETERS___
       {
         "args": [
-          "t2_[a-z0-9]+"
+          "(t2_|a2_)[a-z0-9]+"
```
Were you able to test this in the tag manager?  I see the unit test -- which is super! -- but a test from the UI would be nice confirmation it works as intended.  If you haven't played with tagmanager.google.com before, I can show you.

woodchuckgym commented on template.tpl:
```diff
@@ -910,6 +910,30 @@ scenarios:
 
+    // Verify that the tag finished successfully.
+    assertApi('gtmOnSuccess').wasCalled();
+- name: Test pixel init - set a2_ advertiser ID and integration type
```
TY!

woodchuckgym reviewed (changes requested):
"Requesting changes" for a quick end-to-end test with the newly-committed changes for the business account IDs.

dmaii reviewed (approved):
Agree with Jim but LGTM otherwise

elmiranda commented:
Tested this as a proper tag in the wild, on a test webpage! S/O to Austin who actually has a persistent setup for a GTM tag for exactly this reason. We tested both `t2_33rjacqk` and `a2_bo1kbw5ereg5 `, his account and the test account, respectively. I was able to confirm that the data landed successfully in BQ (no degradation from previous tests).

![image](https://user-images.githubusercontent.com/86022521/214443174-95f86663-b42c-4f06-b333-65549e147b16.png)

woodchuckgym reviewed (approved):
TY!
